testrunner-running/src/main/scala/edu/illinois/cs/testrunner/execution/Utils.java

```diff
@@ -21,7 +27,12 @@
      * full qualified name: com.luojl.demo.JUnit4DemoTest#TestA4
      */
     public static String toFullyQualifiedName(String identifierUniqueId) {
-        Matcher matcher = junit5Pattern.matcher(identifierUniqueId);
+        Matcher matcher = junit5NestedPattern.matcher(identifierUniqueId);
+        if (matcher.find()) {
```

Comment:
What would happen if there are multiple layers of nesting, e.g., Nested1$Nested2$Nested3#test? Is this allowed? Would this regex handle that?

Reply:
Oh, I didn't think about this case. Will try.

Reply:
It turns out the multiple layers of nesting is different, here is an example of two layers:

[engine:junit-jupiter]/[class:com.luojl.demo.InheritedTest]/[nested-class:NestedTest]/[nested-class:DoubleNestedTest]/[method:DoubleNestedTestA()]

There are two `nested-class`

Reply:
I assume the current regex does not handle this general case of multiple nestings? Can you modify the regex and the corresponding matching logic so it outputs the proper format with the $ in between each class?

Reply:
Yea, the current regex can not handle this case. And is it possible for regex to do that? I guess there is some advanced regex rule. I will take a look. Otherwise, I would use an iterated method.

Reply:
Done